application/backend/src/main/resources/apps/typerefinery/components/content/ticker/script.js

```diff
@@ -48,43 +36,166 @@ function updateTickerComponent(tickerData, id) {
+        message.detail,
+        "hello"
+      );
+      var messageText = message.detail;
+      if (typeof message.detail === "object") {
+        messageText = JSON.stringify(message.detail);
+        var messageType = message.detail.type;
+
+        if (messageType === "meta") {
+          var subscribers = message.detail.subscribers;
+          var publishers = message.detail.publish;
+          var calls = message.detail.call;
+
+          // for each object in subscribers add object id to subscribers select element
+          for (var subscriber in subscribers) {
+            let option = document.createElement("option");
+            option.value = subscribers[subscriber];
+            option.text = subscribers[subscriber];
+            document.getElementsByName("subscribers")[0]?.add(option);
+          }
+
+          // for each object in publishers add object id to publishers select element
+          for (var publisher in publishers) {
+            let option = document.createElement("option");
+            option.value = publishers[publisher].id;
+            option.text = publishers[publisher].id;
+            document.getElementsByName("publishers")[0]?.add(option);
+          }
+
+          // for each object in calls add object id to calls select element
+          for (var call in calls) {
+            let option = document.createElement("option");
+            option.value = calls[call].id;
+            option.text = calls[call].id;
+            document.getElementsByName("calls")[0]?.add(option);
+          }
+        }
+      }
+      var messageItem = document.createElement("li");
+      var content = document.createTextNode(messageText);
+      messageItem?.appendChild(content);
+      const payload = message.detail.data.payload;
+      const parsedPayload = JSON.parse(payload);
+      const resultPayload = parsedPayload.data;
+      document.getElementById("messages")?.appendChild(messageItem);
+      updateTickerComponent(resultPayload, component);
+    }
+  );
+  function payload_insert(data) {
+    console.log("payload_insert", data);
+  }
+  // connect to websocket
+  window.MessageService.Client.connect(host, function () {
+    console.log("tms connected cms.");
+    window.MessageService.Client.subscribe("payload_insert", payload_insert);
+  });
+}
+function getDataFromDataSourceJson(jsonUrl,component){
+  const fetchAndUpdateView = async () => {
+    try {
+      const response = await fetch(jsonUrl).then((res) => res.json());
+
+      !response.value
+        ? tickerComponentConnectedViaInitialData(component)
+        : updateTickerComponent(response, component);
+    } catch (error) {
+      tickerComponentConnectedViaInitialData(component)
+    }
+  };
+  fetchAndUpdateView();
+}
+// Update ticker using datasource JSON
+function tickerComponentConnectedViaJSON(jsonUrl, component){
+  // Rendering the template
+  getDataFromDataSourceJson(jsonUrl,component);
```

Comment:
there are nested of function call, so you can make this as one single function

Reply:
Made on single function to update through json things